crates/joystick/src/common.rs

```diff
@@ -0,0 +1,75 @@
+use evdev::KeyCode;
+use std::time::Duration;
+
+pub static TAP_DURATION: u64 = 50;
+
+pub enum KeyAction {
+    Press,
+    Release,
+}
+
+pub trait JoystickInterface {
+    fn press_a(&mut self);
```

Comment:
Possibly, the nicer way here is to set up an `enum`, so it would just be `press(A)` and `release(A)` instead. `A` (and the other buttons) would map to the internal types. Further, we can have another mapping in the game-specific code to map another `enum` holding the various actions (`Confirm`, `Cancel`, `Graplou` etc.) to the buttons.